src/mudlet.cpp

```diff
@@ -2417,3 +2423,73 @@ void mudlet::requestProfilesToReloadMaps( QList<QString> affectedProfiles )
 {
     emit signal_profileMapReloadRequested( affectedProfiles );
 }
+
+void mudlet::slot_gamepadButtonPress(int deviceId, QGamepadManager::GamepadButton button, double value)
+{
+    Host * pH = getActiveHost();
+    if( ! pH ) return;
+    TEvent event;
+    event.mArgumentList.append( "sysGamepadButtonPress" );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append( QString::number(deviceId) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    event.mArgumentList.append( QString::number(button) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    event.mArgumentList.append( QString::number(value) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    pH->raiseEvent( event );
+}
+
+void mudlet::slot_gamepadButtonRelease(int deviceId, QGamepadManager::GamepadButton button)
+{
+    Host * pH = getActiveHost();
+    if( ! pH ) return;
+    TEvent event;
+    event.mArgumentList.append( "sysGamepadButtonRelease" );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append( QString::number(deviceId) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    event.mArgumentList.append( QString::number(button) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    pH->raiseEvent( event );
+}
+
+void mudlet::slot_gamepadConnected(int deviceId)
+{
+    Host * pH = getActiveHost();
+    if( ! pH ) return;
+    TEvent event;
+    event.mArgumentList.append( "sysGamepadConnected" );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append( QString::number(deviceId) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    pH->raiseEvent( event );
```

Comment:
There is supposed to be, but it seems like a lot of the other parts of the kit don't work properly. For instance, if I grab the deviceId of one of the gamepads and make an object to manipulate that qGamepad with, I can't get basic things like its method `bool	buttonStart() const
` functioning properly. I was also unable too get device info when I tried reading the name of the device on a standalone test app I made. Maybe it's just a limitation in Windows or something, but I wasn't having too much luck and didn't see it being too useful for Mudlet purposes.